examples/seq2seq/seq2seq_trainer.py

```diff
@@ -114,23 +115,22 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
                 else DistributedSampler(self.train_dataset)
             )
 
-    def compute_loss(self, model, inputs):
-        labels = inputs.pop("labels")
-        outputs = model(**inputs, use_cache=False)
-        logits = outputs[0]
-        return self._compute_loss(logits, labels)
-
-    def _compute_loss(self, logits, labels):
+    def _compute_loss(self, model, inputs):
         if self.args.label_smoothing == 0:
-            # Same behavior as modeling_bart.py
-            loss_fct = torch.nn.CrossEntropyLoss(ignore_index=self.config.pad_token_id)
```

Comment:
This does not seem to work for all models (`EncoderDecoderModel` does not work with it) -> Let's instead use the loss function of each model here.

Reply:
loss functions of model use  -100 as `ignore_index` , we will also need to replace `pad` tokens in `labels` with -100

Reply:
I usually do this manually before -> should that be the role of the `Seq2SeqTrainer`? `Trainer` also does not have this feature

Reply:
Ignoring pad_token_id confused lots of people and helps metrics so we automated it. 
Related: #7828

Reply:
> I usually do this manually before

we could do this in the `collator`, but we won't need to do if #7828 is merged

Reply:
we will still need FSMT/T5.

Reply:
PyTorch's CE loss function has `-100` as a default value and from what I understood it is the default behavior of the library to ignore tokens when there have the index -100 and not when there are equal to the padding token (often we set padding token == -100): https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

It would require models to manually replace tokens with -100, but I think that's how it should be done in general in the library. How would be handle models that don't have a padding_token or want to disregard loss of more than just the padding token? For such cases I think it can be quite handy if the user overwrites all labels he does not want to consider with -100

Reply:
we will discuss on zoom!